packages/shared/src/lib/core/network/classes/isc-chain.class.ts

```diff
@@ -47,9 +51,16 @@ export class IscChain extends BaseEvmNetwork {
         return (await response.json()) as IIscChainMetadata
     }
 
-    async getLatestBlock(): Promise<IBlock> {
-        const number = await this.provider.eth.getBlockNumber()
-        return this.provider.eth.getBlock(number)
+    async getBalance(account: IAccountState): Promise<ITokenBalance | undefined> {
+        const evmAddress = account.evmAddresses?.[this.coinType]
+        if (!evmAddress) {
+            return undefined
+        }
+
+        const tokenBalance = (await super.getBalance(account)) ?? {}
+        const iscBalance = (await fetchIscAssetsForAccount(getActiveProfileId(), evmAddress, this, account)) ?? {}
```

Comment:
You should inject the profile/profileId as a dependency to the constructor and move the polling of the balance also in this class, so we can clear it when the chain is destroyed.